weave-js/src/util/events.ts

```diff
@@ -34,9 +34,38 @@ export const setupAnalytics = () => {
 
 setupAnalytics();
 
-/**
- * TODO: Connect to router.
- */
 export function trackPage(properties: object, options: object) {
   (window.analytics as any)?.page?.(properties, options);
 }
+
+export function trackEvent(
+  eventName: string,
+  eventData: Record<string, unknown>
+) {
+  // Taken from W&B app's Analytics.
+  if (
+    Array.isArray(eventData) || // Segment accepts lists but won't process them
+    typeof eventData !== 'object' || // exclude the primitives
+    eventData === null // typeof null => 'object' :grimace:
+    // note this still leaves holes for Date / Error / any other weird "object" types
+  ) {
+    throw new Error('Analytics data must be a non-array object.');
+  }
+
+  (window.analytics as any)?.track?.(eventName, eventData);
+}
+
+export function trackNewBlankBoardClicked(source: string) {
```

Comment:
My only comment here is that i believe each of these events will create their own table - which could be a pain to keep track off and informally named. Consider using one table: `weave_app_event` and then having the specific details of the action differentiated in the action payload

Reply:
nice. makes sense -- will revise.